src/quicklist.c

```diff
@@ -1019,7 +1016,8 @@ REDIS_STATIC void _quicklistInsert(quicklist *quicklist, quicklistEntry *entry,
     }
 
     if(unlikely(isLargeElement(sz))) {
-        if (QL_NODE_IS_PLAIN(node)) {
+        if (QL_NODE_IS_PLAIN(node) || (full && ((at_tail && !avail_next && after) ||
+                                               (at_head && !avail_prev && !after)))) {
```

Comment:
i think `full` and `avail_next` are guaranteed to be 1 and 0 since `_quicklistNodeAllowInsert` has this check:
```c
    if(unlikely(QL_NODE_IS_PLAIN(node) || isLargeElement(sz)))
        return 0;
```
am i missing anything?